### ISSUE TITLE
No colour output with groff 1.23.0

### ISSUE THREAD

**Issue by pezz** [jedsoft/most#18]
Hi there,

Running Arch Linux and ```most``` no longer produces colours in its man page output with groff 1.23.0 installed.

If I downgrade to groff 1.22.4 it works as expected.

Happy to provide further info if I can.

Cheers.

**Comment by pezz** [comment]
```export GROFF_NO_SGR=1```

Returns most to its colourful niceness.

Sorry for the noise.

**Comment by kavehtehrani** [comment]
> `export GROFF_NO_SGR=1`
> 
> Returns most to its colourful niceness.
> 
> Sorry for the noise.

This fixed my issue. Thank you.

**Comment by jedsoft** [comment]
I investigated this a bit on my debian system which uses groff version
1.22.4.  In particular, I did the following to look at the output of a
man page (grotty) in an editor (`jed`):

  GROFF_SGR=1 PAGER=jed man grotty

This showed the various escape sequences embedded in the man page,
e.g.,

 By default, ^[[1mgrotty ^[[22memits SGR escape sequences (from ISO
 6429, also called ANSI color escapes) to change text attributes
 (bold, italic, colors).  This makes it possible to have eight
 different background and foreground colors; additionally, bold and
 italic attributes can be used ^[[4m^[[1mat^[[24m ^[[4mthe^[[24m
 ^[[4msame^[[24m ^[[4mtime^[[24m ^[[22m(by using the BI font).

The above contains the following escape sequences:

  ^[[1m   (bold)
  ^[[22m  (normal)
  ^[[4m   (underline)
  ^[[24m  (turn off underline)

The abilty for those to be rendered depends upon your terminal.  Xterm
renders them but urxvt does not.

Nowhere in the man page did I see any escape sequences that pertained
to colors.

Playing the same game with:

  GROFF_NO_SGR=1 PAGER=jed man grotty

produced:

 By default, g^Hgr^Hro^Hot^Htt^Hty^Hy emits SGR escape sequences (from
 ISO 6429, also called ANSI color escapes) to change text attributes
 (bold, italic, colors).  This makes it possible to have eight
 different background and foreground colors; additionally, bold and
 italic attributes can be used _^Ha^Ha_^Ht^Ht _^Ht^Ht_^Hh^Hh_^He^He
 _^Hs^Hs_^Ha^Ha....

Here, ^H is a backspace character which `most` will interpret as
overstrike.  So "g^Hg" represents a bold "g", and "_^Ha" represents an
underlined "a".  When `most` sees such sequences, it maps those to the
colors defined in your .mostrc file.

In conclusion, I do not believe that `most` is at fault here.

Thanks for the followup.
--John

On Sun, 12 Nov 2023 07:00:45 -0800, Kaveh ***@***.***> said:
>> `export GROFF_NO_SGR=1`
>>
>> Returns most to its colourful niceness.
>>
>> Sorry for the noise.
>
>This fixed my issue. Thank you.
>

**Comment by edrozenberg** [comment]
As described by this Arch topic (https://bbs.archlinux.org/viewtopic.php?id=287185), you may want to instead use `export MANROFFOPT=-c` which limits the scope of the effect on groff processing to only manpages.
